include/swift/AST/DiagnosticEngine.h

```diff
@@ -629,6 +634,23 @@ namespace swift {
     DiagnosticState(DiagnosticState &&) = default;
     DiagnosticState &operator=(DiagnosticState &&) = default;
   };
+
+  class LocalizationProducer {
+  public:
+    std::vector<DiagnosticNode> diagnostics;
+    explicit LocalizationProducer(std::string locale, std::string path);
+    
+    /// If the  message isn't available/localized in the current `yaml` file,
+    /// return the message from the `.def` file.
```

Comment:
Nit: Instead of concretizing `.def` file here, let's just say `return fallback default message`.